cmd/operator-sdk/add/api.go

```diff
@@ -84,19 +142,144 @@ Example:
 	}
 	apiCmd.Flags().BoolVar(&skipGeneration, "skip-generation", false,
 		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
+	apiCmd.Flags().StringVar(&helmChartRef, "helm-chart", "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path)")
+	apiCmd.Flags().StringVar(&helmChartVersion, "helm-chart-version", "",
+		"Specific version of the helm chart (default is latest versio)")
+	apiCmd.Flags().StringVar(&helmChartRepo, "helm-chart-repo", "",
+		"Chart repository URL for the requested helm chart")
 
 	return apiCmd
 }
 
 func apiRun(cmd *cobra.Command, args []string) error {
+
 	projutil.MustInProjectRoot()
+	log.Infof("Generating api version %s for kind %s.", apiVersion, kind)
+
+	operatorType := projutil.GetOperatorType()
 
-	// Only Go projects can add apis.
-	if err := projutil.CheckGoProjectCmd(cmd); err != nil {
+	switch operatorType {
+	case projutil.OperatorTypeGo:
+		if err := doGoAPIScaffold(); err != nil {
+			return err
+		}
+	case projutil.OperatorTypeAnsible:
+		if err := doAnsibleAPIScaffold(); err != nil {
+			return err
+		}
+	case projutil.OperatorTypeHelm:
+		if err := doHelmAPIScaffold(); err != nil {
+			return err
+		}
+	}
```

Comment:
Need to check for `projutil.OperatorTypeUnknown` here and error out.

Reply:
Since we are checking for operatortype, prior to switch case, This is  taken care of.
	`flagSet.OperatorType = projutil.GetOperatorType()`

Reply:
Since we're removing the `type` check from verify flags, I think this switch will end up needing a case for `projutil.OperatorTypeUnknown`

Reply:
I'm seeing the following on this branch:

```console
$ find .
.
./build
./build/Dockerfile

$ operator-sdk add api
INFO[0000] Generating api version  for kind .
INFO[0000] API generation complete.
```

We should actually check right after `operatorType := projutil.GetOperatorType()` so that we don't log `Generating api version ...`

```go
operatorType := projutil.GetOperatorType()
if operatorType == projutil.OperatorTypeUnknown {
	return projutil.ErrUnknownOperatorType{}
}
```